client/java/src/main/java/io/openlineage/client/dataset/Naming.java

```diff
@@ -0,0 +1,797 @@
+      this.table = checkArgumentNotEmpty(table, "table");
+    }
+
+    @Override
+    public String getNamespace() {
+      return "teradata://" + host + ":" + port;
+    }
+
+    @Override
+    public String getName() {
+      return database + "." + table;
+    }
+  }
+
+  /** Naming implementation for Redshift. */
+  public static class RedshiftNaming implements DatasetNaming {
+    private final String cluster_identifier;
+    private final String region_name;
+    private final String port;
+    private final String database;
+    private final String schema;
+    private final String table;
+
+    public RedshiftNaming(
+        String cluster_identifier,
+        String region_name,
+        String port,
+        String database,
+        String schema,
+        String table) {
+      this.cluster_identifier = checkArgumentNotEmpty(cluster_identifier, "cluster_identifier");
+      this.region_name = checkArgumentNotEmpty(region_name, "region_name");
+      this.port = checkArgumentNotEmpty(port, "port");
+      this.database = checkArgumentNotEmpty(database, "database");
+      this.schema = checkArgumentNotEmpty(schema, "schema");
+      this.table = checkArgumentNotEmpty(table, "table");
+    }
+
+    @Override
+    public String getNamespace() {
+      return "redshift://" + cluster_identifier + "." + region_name + ":" + port;
+    }
+
+    @Override
+    public String getName() {
+      return database + "." + schema + "." + table;
+    }
+  }
+
+  /** Naming implementation for Snowflake. */
+  public static class SnowflakeNaming implements DatasetNaming {
+    private final String organization_name;
+    private final String account_name;
+    private final String database;
+    private final String schema;
+    private final String table;
+
+    public SnowflakeNaming(
+        String organization_name,
+        String account_name,
+        String database,
+        String schema,
+        String table) {
+      this.organization_name = checkArgumentNotEmpty(organization_name, "organization_name");
+      this.account_name = checkArgumentNotEmpty(account_name, "account_name");
+      this.database = checkArgumentNotEmpty(database, "database");
+      this.schema = checkArgumentNotEmpty(schema, "schema");
+      this.table = checkArgumentNotEmpty(table, "table");
+    }
+
+    @Override
+    public String getNamespace() {
+      return "snowflake://" + organization_name + "-" + account_name;
+    }
+
+    @Override
+    public String getName() {
+      return database + "." + schema + "." + table;
+    }
+  }
+
+  /** Naming implementation for Trino. */
+  public static class TrinoNaming implements DatasetNaming {
+    private final String host;
+    private final String port;
+    private final String catalog;
+    private final String schema;
+    private final String table;
+
+    public TrinoNaming(String host, String port, String catalog, String schema, String table) {
```

Comment:
Constructors having multiple arguments of the same type are error-prone. 
`openlineage-java` already contains Lombok dependency. How about using `Builder` ?